version-rules.xml

```diff
@@ -0,0 +1,16 @@
+<ruleset comparisonMethod="maven"
+         xmlns="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0 http://mojo.codehaus.org/versions-maven-plugin/xsd/rule-2.0.0.xsd">
+    <ignoreVersions>
+        <!--only release builds-->
+        <ignoreVersion type="regex">.*(alpha|ALPHA|beta|BETA|b|rc|RC|M).*</ignoreVersion>
+        
+        <!--ignore some specific version strings which don't follow the common version format-->
+        <ignoreVersion type="exact">20020829</ignoreVersion>
```

Comment:
What is the purpose of these ignoreVersions?

Reply:
version-rules.xml is for the version-maven-plugin* i added to the main pom.xml.

the plugin can be run manually with "mvn versions:display-dependency-updates" and will list all dependencies in this project which could be updated (just for convenience, the command itself does not update anything - but it could be used for that too).

The ignoreVersion regex tries to filter out any recommendations which aren't release versions. The remaining rules in that file blacklist some old artifact snapshots.

the maven-antrun-plugin for example has an old artifact with the version 20020829, which would be always recommended by the plugin if not blacklisted since it would read it as very high version number. There might be a better way of doing this but luckily it needed only a handful of entries on the blacklist.

*https://www.mojohaus.org/versions-maven-plugin/index.html